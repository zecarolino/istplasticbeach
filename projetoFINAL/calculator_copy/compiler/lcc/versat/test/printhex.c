int x = 0xDB97531, vec[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };
void putchar(int ch) { asm("\trdw R11\n\taddi 2\n\twrw RB\n\trdwb\n\trdwb\n\twrw CPRT_BASE\n"); }
int main() {
	do
		putchar(vec[x & 0xF]);
	while (x >>= 4);
	putchar('\n');
	return 0;
}
