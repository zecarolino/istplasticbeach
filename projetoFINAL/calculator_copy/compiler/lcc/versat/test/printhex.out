VCD info: dumpfile xtop.vcd opened for output.
13579BD
0
Catched Trap at time  1406 clock cycles:
